app/app.go

```diff
@@ -1540,6 +1522,35 @@ func (app *App) ProcessBlock(ctx sdk.Context, txs [][]byte, req BlockProcessRequ
 	return events, txResults, endBlockResp, nil
 }
 
+func (app *App) DecodeTransactionsConcurrently(ctx sdk.Context, txs [][]byte) []sdk.Tx {
+	typedTxs := make([]sdk.Tx, len(txs))
+	wg := sync.WaitGroup{}
+	for i, tx := range txs {
+		wg.Add(1)
+		go func(idx int, encodedTx []byte) {
+			defer wg.Done()
+			typedTx, err := app.txDecoder(encodedTx)
+			// get txkey from tx
+			if err != nil {
+				ctx.Logger().Error(fmt.Sprintf("error decoding transaction at index %d due to %s", idx, err))
+				typedTxs[idx] = nil
+			} else {
+				if isEVM, _ := evmante.IsEVMMessage(typedTx); isEVM {
+					msg := evmtypes.MustGetEVMTransactionMessage(typedTx)
+					if err := evmante.Preprocess(ctx, msg); err != nil {
+						ctx.Logger().Error(fmt.Sprintf("error preprocessing EVM tx due to %s", err))
+						typedTxs[idx] = nil
```

Comment:
Will this trigger nil pointer exception later on and cause chain to halt?

Reply:
this is existing behavior. And i think it's handled here https://github.com/sei-protocol/sei-chain/blob/seiv2/app/app.go#L1336-L1341